### PR TITLE
Clarify how read! works

### LINKED ISSUE #185: how does read! work?
How does the user get the indices of the variables and constraints loaded using `read!`?

### PR DESCRIPTION
**Pull request by odow**
Closes #185

### CODE REVIEW

File: src/MathOptInterface.jl [hunk 1]
```diff
@@ -51,6 +51,13 @@ Read the file `filename` into the instance `instance`. If `m` is non-empty, this
 throw an error.
 
 Supported file types depend on the instance type.
+
+### Note
+
+Once the model is loaded into the model, users can query the variables via
+`get(model, ListOfVariableIndices())`. However, this could return the variables
+in any order. Therefore, users should look up up each variable index by name:
```
Comment (mlubin) [inline src/MathOptInterface.jl]: MPS files have a well-defined order on the variables. Would we still expect `ListOfVariableIndices()` to return them back in an arbitrary order in that case?

Comment (odow) [reply src/MathOptInterface.jl]: Presumably. But that seems a little flakey.

Comment (odow) [reply src/MathOptInterface.jl]: This is another manual doc then? Recommending `get.(model, VariableIndex, ["x", "y", "z"])` without mentioning `ListOfVariableIndices`?

File: src/MathOptInterface.jl [hunk 2]
```diff
@@ -51,6 +51,13 @@ Read the file `filename` into the instance `instance`. If `m` is non-empty, this
 throw an error.
 
 Supported file types depend on the instance type.
+
+### Note
+
+Once the model is loaded into the model, users can query the variables via
```
Comment (blegat) [inline src/MathOptInterface.jl]: "Once the model is loaded into the model" is a bit difficult to grasp :-P Maybe "Once `filename`'s model is loaded in to `model`" reads better

Comment (mlubin) [reply src/MathOptInterface.jl]: "Once the contents of the file are loaded into the model" maybe

**Comment by odow** [comment]
Updated to reflect comments.

File: src/MathOptInterface.jl [hunk 3]
```diff
@@ -51,6 +51,14 @@ Read the file `filename` into the model `model`. If `m` is non-empty, this may
 throw an error.
 
 Supported file types depend on the model type.
+
+### Note
+
+Once the contents of the file are loaded into the model, users can query the variables via
+`get(model, ListOfVariableIndices())`. However, some filetypes, such as LP files, do not
+maintain an explicit ordering of the variables. Therefore, the returned list may be in an
+arbitrary order. To guarantee the ordering of the returned indices, users should look up
```
Comment (mlubin) [inline src/MathOptInterface.jl]: "To avoid depending on the order of the indices, users should look up ..."

**Pull request merged**